src/main/scala/firrtl/Compiler.scala

```diff
@@ -373,6 +373,9 @@ trait Compiler extends LazyLogging {
     */
   def transforms: Seq[Transform]
 
+  require(transforms.size >= 1,
+          s"Compiler transforms for '${this.getClass.getName}' was empty. A compiler must have at least ONE Transform!")
```

Comment:
It would be backwards incompatible, but we could consider changing the type of transforms to `::[Transform]` and then that would enforce non-empty transforms in the type system.

Reply:
This is way cleaner (and I had no idea you could do that!). I would like to break it, but I defer to you re: backwards incompatible changes.

Reply:
I propose that we kick this down the road, potentially for more discussion. I think that the Dependency API may deprecate the Compiler. It's likely not worth it to introduce multiple backwards incompatible changes for the purposes of type safety.